components/builder-graph/src/acyclic_package_graph.rs

```diff
@@ -234,6 +246,153 @@ impl PackageGraphTrait for AcyclicPackageGraph {
                 connected_comp: connected_components(&self.graph),
                 is_cyclic:      is_cyclic_directed(&self.graph), }
     }
+
+    // compute_build for AcyclicGraph
+    // Process
+    // 1) Take kernel of packages, and recursively expand it over reverse build/runtime deps
+    //    creating a new sub-graph.
+    // 2) Filter unbuildable nodes in subgraph, flag them as directly
+    //    (no plan connection, autobuilds off) unbuildable
+    // 3) Use the unbuildable packages as kernel, expand it over the subgraph to
+    //    create a list of all unbuildable packages.
+    // 4) Iterate over all unbuildable packages.
+    //    Flag new entries as indirectly unbuildable.
+    //    Remove that package from the sub-graph
+    // 5) Walk the remaining sub-graph, collecting dependencies that are external to
+    //    the sub-graph.
+    // 6) Create a PackageBuildManifestGraph using the information generated in 1-5.
+    fn compute_build(&self,
+                     touched: &[PackageIdentIntern],
+                     unbuildable: &dyn Unbuildable,
+                     target: PackageTarget /* WRONG WRONG WRONG WRONG, WRONG WRONG WRONG
+                                            * WRONG */)
+                     -> PackageBuildManifest {
```

Comment:
I'm curious if you weighed the effort vs benefits of having `compute_build` instead return a `Result<PackageBuildManifest`? There's an `unwrap()`, an `.expect()` and two `panic!` calls in the code path of this function which leads me to think that it has the potential to be fallible and our future selves may be better served handling the failure as opposed to taking an outage if a service crashes.

Reply:
Also, it may prove useful to add a `TODO` around `PackageTarget` so that the uneducated reader can understand why this is so wrong :smiley:

Reply:
> Also, it may prove useful to add a `TODO` around `PackageTarget` so that the uneducated reader can understand why this is problematic smiley

For context, this is a bit of left over from the previous implementation when target was added late.  The wrongness of it came from the fact that the graph has a target implicitly as part of it, but we still had to thread that value through in order to constrain queries to the database about the package. On our original pass through, we were hesitant to remove it, given it was in legacy code that we intend to remove in the near future.  However, as we revisited it for this review, we realized it would be a very small change to make the PackageTarget part of the struct. 

Fixed in 48ba35769a9de934379d165377e8cc9d0c708333